README.md

```diff
@@ -76,6 +76,12 @@ addEventListener("message", async ({data}) => {
 });
 ```
 
+## `ModuleBlock` object model
+
+A module block expression `module { }` evaluates to an instance of a `ModuleBlock` class. Instances are frozen, and the same instance is returned each time for a particular Parse Node, like tagged template literals. `Module` instances have an internal slot [[ModuleBlockHostData]]. This internal slot is filled in by the host in a new host hook called when parsing each module block. (Alternative: A fresh, mutable `ModuleBlock` is returned each time the `module { }` is evaluated, with the same [[ModuleBlockHostData]] each time.) In `import()`, if the parameter has a [[ModuleBlockHostData]] internal slot, it is passed up as is to the dynamic import host hook, and ToString is only called on other values without this slot. 
```

Comment:
Question: Which, if any, of these assertions throw?

```js
const arr = new Array(2);
for(i = 0; i < arr.length; i++) {
  arr[i] = module {export default 4};
}
console.assert(arr[0] == arr[1]); // ?
console.assert(await import(arr[0]) == await import(arr[1])); // ?

const m1 = module {export default 4};
const m2 = module {export default 4};
console.assert(m1 == m2); // ?
console.assert(await import(m1) == await import(m2)); // ?

```

Reply:
(and should we add this code sample to clarify?)

Reply:
With the main option, the last two asserts would throw. With the alternate, the first assertion would throw as well. But all agree that the second assertion does not throw.

Code samples could be great here. Do you think you could push the appropriate clarifications to all of this?

Reply:
Strong preference for no caching behavior, i.e. each `module { }` evaluation resulting in a new instance.